src/algorand_transactions/transactions_group.rs

```diff
@@ -0,0 +1,220 @@
+use serde::{Deserialize, Serialize, Serializer};
+use sha2::Digest;
+
+use crate::{
+    algorand_errors::AlgorandError,
+    algorand_types::{Byte, Bytes, Result},
+    AlgorandHash,
+    AlgorandSignedTransaction,
+    AlgorandTransaction,
+};
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize)]
+pub struct TxGroup {
+    #[serde(rename = "txlist", default)]
+    tx_group_hashes: Vec<AlgorandHash>,
+}
+
+impl TxGroup {
+    const MAX_TX_GROUP_SIZE: usize = 16;
+
+    pub fn new(tx_group_hashes: Vec<AlgorandHash>) -> TxGroup {
+        TxGroup { tx_group_hashes }
+    }
+
+    pub fn assign_group_id(txns: &mut [&mut AlgorandTransaction]) -> Result<AlgorandHash> {
+        let gid = TxGroup::compute_group_id(txns)?;
+        for tx in txns {
+            tx.assign_group_id(gid);
+        }
+        Ok(gid)
+    }
+
+    pub(crate) fn compute_group_id(txns: &[&mut AlgorandTransaction]) -> Result<AlgorandHash> {
+        if txns.is_empty() {
+            return Err(AlgorandError::Custom("Empty transactions list".to_string()));
+        }
+        if txns.len() > Self::MAX_TX_GROUP_SIZE {
+            return Err(AlgorandError::Custom("Too much transaction".to_string()));
+        }
+        let mut ids: Vec<AlgorandHash> = vec![];
+        for t in txns {
+            ids.push(t.to_raw_tx_id()?);
+        }
```

Comment:
Let's try and be as explicit with naming as possible to make the lives of future readers easier:

```suggestion
        let mut tx_ids: Vec<AlgorandHash> = vec![];
        for tx in txns {
            tx_ids.push(tx.to_raw_tx_id()?);
        }
```

Also, there's no need for a mutable variable and a loop here. Just map over the txs:

```rust
let ids = txns.iter().map(AlgorandTransaction::to_raw_tx_id).collect::<Result<Vec<AlgorandHash>>>()?;
```

Reply:
Compiler does not give me to pass `AlgorandTransaction::to_raw_tx_id` to `map`, and I think it is because `to_raw_tx_id` needs to be applied to a struct.
For instance, this works:
```
        let tx_ids = txns.iter().map(|tx| tx.to_raw_tx_id()).collect::<Result<Vec<AlgorandHash>>>()?;
```

Reply:
^ no worries! It's doing the exact same thing, I just thought we might be able to call it first class for less visual noise, but alas!